src/ehttpc.erl

```diff
@@ -271,6 +275,39 @@ gun_opts([{transport_opts, TransportOpts} | Opts], Acc) ->
 gun_opts([_ | Opts], Acc) ->
     gun_opts(Opts, Acc).
 
+maybe_ipv6(Host, Port, #{transport_opts := TransportOpts} = Opts) ->
+    Opts#{transport_opts := maybe_ipv6_1(Host, Port, TransportOpts)};
+maybe_ipv6(_Host, _Port, Opts) ->
+    Opts.
+
+maybe_ipv6_1(Host, Port, Opts) ->
+    case lists:member(inet, Opts) orelse lists:member(inet6, Opts) of
+        true -> Opts; %% user has made the decision
+        false -> maybe_ipv6_2(Host, Port, Opts)
+    end.
+
+maybe_ipv6_2(Host, _Port, Opts) when is_tuple(Host) ->
+    %% ip tuple provided
+    case tuple_size(Host) of
+        4 -> [inet | Opts];
+        8 -> [inet6 | Opts]
+    end;
+maybe_ipv6_2(Host, Port, Opts) ->
+    case inet:parse_address(Host) of
+        {ok, Ip} -> maybe_ipv6_2(Ip, Port, Opts); %% ip string provided
+        {error, einval} -> maybe_ipv6_3(Host, Port, Opts)
+    end.
+
+maybe_ipv6_3(Host, Port, Opts) ->
+    case gen_tcp:connect(Host, Port, [inet6], 5000) of
```

Comment:
Maybe `inet:getaddr/2` is more appropriate?

Reply:
We have to prob the port.
E.g. localhost is reachable from both v4 and v6
But the service may only listening on one stack